WalletWasabi.Fluent/ViewModels/Wallets/Send/LabelSelectionViewModel.cs

```diff
@@ -275,21 +277,38 @@ private void OnSelectionChanged()
 			_hiddenIncludedPockets.Add(_semiPrivatePocket);
 		}
 
-		var totalSelected = sumOfWhiteList + _hiddenIncludedPockets.Sum(x => x.Amount);
+		var totalSelected = whiteListPockets.EffectiveSumValue(_feeRate) + _hiddenIncludedPockets.EffectiveSumValue(_feeRate);
 
 		EnoughSelected = totalSelected >= _targetAmount;
 
 		this.RaisePropertyChanged(nameof(LabelsWhiteList));
 		this.RaisePropertyChanged(nameof(LabelsBlackList));
 	}
 
-	private bool IsPrivatePocketNeeded() =>
-		(NonPrivatePockets.Sum(x => x.Amount) < _targetAmount && _privatePocket.Amount + _semiPrivatePocket.Amount < _targetAmount && NonPrivatePockets.Sum(x => x.Amount) + _privatePocket.Amount >= _targetAmount) ||
-		(LabelsWhiteList.IsEmpty() && _privatePocket.Amount >= _targetAmount);
+	private bool IsPrivatePocketNeeded()
```

Comment:
Note that, this and the next method seems to be conceptually wrong, but it doesn't seem to be changing the existing behavior so this is not a critique of this PR.

Reply:
What do you mean by that? There are tests for them that ensure they are correct.

Reply:
https://github.com/zkSNACKs/WalletWasabi/issues/9092#issuecomment-1247764567